services/api/service.go

```diff
@@ -1519,50 +1738,76 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 	}
 
 	// Simulate the block submission and save to db
-	fastTrackValidation := builderIsHighPrio && bidIsTopBid
+	fastTrackValidation := builderEntry.status.IsHighPrio && bidIsTopBid
 	timeBeforeValidation := time.Now().UTC()
+
 	log = log.WithFields(logrus.Fields{
 		"timestampBeforeValidation": timeBeforeValidation.UTC().UnixMilli(),
 		"fastTrackValidation":       fastTrackValidation,
 	})
 
-	validationRequestPayload := &common.BuilderBlockValidationRequest{
-		BuilderSubmitBlockRequest: *payload,
-		RegisteredGasLimit:        slotDuty.Entry.Message.GasLimit,
-	}
-	requestErr, validationErr = api.blockSimRateLimiter.Send(req.Context(), validationRequestPayload, builderIsHighPrio, fastTrackValidation)
-	validationDurationMs := time.Since(timeBeforeValidation).Milliseconds()
-	log = log.WithFields(logrus.Fields{
-		"timestampAfterValidation": time.Now().UTC().UnixMilli(),
-		"validationDurationMs":     validationDurationMs,
-	})
-
-	if requestErr != nil {
-		// Request error -- log and return error
-		log.WithFields(logrus.Fields{
-			"requestErr": requestErr.Error(),
-			"durationMs": validationDurationMs,
-		}).Info("block validation failed - request error")
-		if os.IsTimeout(requestErr) {
-			api.RespondError(w, http.StatusGatewayTimeout, "validation request timeout")
-		} else {
-			api.RespondError(w, http.StatusBadRequest, requestErr.Error())
-		}
-		return
+	nextTime = time.Now().UTC()
+	pf.Prechecks = uint64(nextTime.Sub(prevTime).Microseconds())
+	prevTime = nextTime
+
+	// Construct simulation request.
+	opts := blockSimOptions{
+		isHighPrio: builderEntry.status.IsHighPrio,
+		fastTrack:  fastTrackValidation,
+		log:        log,
+		builder:    builderEntry,
+		req: &common.BuilderBlockValidationRequest{
+			BuilderSubmitBlockRequest: *payload,
+			RegisteredGasLimit:        slotDuty.Entry.Message.GasLimit,
+		},
+	}
+	// With sufficient collateral, process the block optimistically.
+	if builderEntry.status.IsOptimistic &&
+		builderEntry.collateral.Cmp(payload.Value()) >= 0 &&
+		payload.Slot() == api.optimisticSlot.Load() {
```

Comment:
just a sidenote -- this will not allow any optimistic block submissions after a missed slot (because headSlot won't be updated, which is why we removed the general check earlier on)

Reply:
right, i think this is good actually! we want optimistic to be as safe of an operation as possible and if builders are building on different root and/or there are missing slots, we want to default to the safest mode of operation. we didnt design it specifically for missing slots, but IMO this is WAI